<h2>Why is this an issue?</h2>
<p>This rule applies whenever an <code>if</code> statement is followed by one or more <code>else if</code> statements; the final <code>else if</code>
should be followed by an <code>else</code> statement.</p>
<p>The requirement for a final <code>else</code> statement is defensive programming.</p>
<p>The <code>else</code> statement should either take appropriate action or contain a suitable comment as to why no action is taken. This is
consistent with the requirement to have a final <code>default</code> clause in a <code>switch</code> statement.</p>
<h3>Noncompliant code example</h3>
<pre data-diff-id="1" data-diff-type="noncompliant">
if x == 0 {
    do_something();
} else if x == 1 {
    do_something_else();
}
</pre>
<h3>Compliant solution</h3>
<pre data-diff-id="1" data-diff-type="compliant">
if x == 0 {
    do_something();
} else if x == 1 {
    do_something_else();
} else {
    panic!("Unexpected value for x");
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#else_if_without_else">https://rust-lang.github.io/rust-clippy/master/index.html#else_if_without_else</a> </li>
</ul>

