<h2>Why is this an issue?</h2>
<p>Declaring a variable only to immediately return it is considered a bad practice because it adds unnecessary complexity to the code. This practice
can make the code harder to read and understand, as it introduces an extra step that doesnâ€™t add any value. Instead of declaring a variable and then
immediately returning, it is generally better to return or throw the value directly. This makes the code cleaner, simpler, and easier to
understand.</p>
<h2>How to fix it</h2>
<p>Declaring a variable only to immediately return it is considered a bad practice because it adds unnecessary complexity to the code. To fix the
issue, return the value directly.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
fn compute_duration_in_milliseconds(hours: u32, minutes: u32, seconds: u32) -&gt; u32 {
    let duration = (((hours * 60) + minutes) * 60 + seconds) * 1000;
    duration
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
fn compute_duration_in_milliseconds(hours: u32, minutes: u32, seconds: u32) -&gt; u32 {
    (((hours * 60) + minutes) * 60 + seconds) * 1000
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#/let_and_return">https://rust-lang.github.io/rust-clippy/master/index.html#/let_and_return</a> </li>
</ul>

