<h2>Why is this an issue?</h2>
<p>Using the same value on either side of a binary operator is almost always a mistake. In the case of logical operators, it is either a copy/paste
error and therefore a bug, or it is simply wasted code, and should be simplified. In the case of bitwise operators and most binary mathematical
operators, having the same value on both sides of an operator yields predictable results, and should be simplified.</p>
<h3>Noncompliant code example</h3>
<pre>
if a == a { // always true
  doZ();
}
if a != a  { // always false
  doY();
}

assert_eq!(a, a); // always true
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#eq_op">https://rust-lang.github.io/rust-clippy/master/index.html#eq_op</a> </li>
</ul>

