<h2>Why is this an issue?</h2>
<p>Calling the <code>to_string</code> method on a value that is already of type <code>String</code> in Rust is redundant and unnecessary. This
redundant call does not change the value and only adds unnecessary complexity to the code. It can also be confusing to readers and may lead to
misunderstandings about the codeâ€™s intent.</p>
<p>If the intent was to clone the string, it is better to use the <code>clone</code> method instead.</p>
<p>Avoiding such redundant calls helps keep the code clean, concise, and more readable.</p>
<h3>Noncompliant code example</h3>
<pre data-diff-id="1" data-diff-type="noncompliant">
let message = String::from("hello world");
println!("{}", message.to_string()); // Noncompliant: 'message' is already a String
</pre>
<h3>Compliant solution</h3>
<pre data-diff-id="1" data-diff-type="compliant">
let message = String::from("hello world");
println!("{}", message);

// or

let cloned_message = message.clone(); // Use 'clone' if you need a duplicate
println!("{}", cloned_message);
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#string_to_string">https://rust-lang.github.io/rust-clippy/master/index.html#string_to_string</a> </li>
</ul>

