<h2>Why is this an issue?</h2>
<p>A chain of <code>if</code>/<code>else if</code> statements is evaluated from top to bottom. At most, only one branch will be executed: the first
one with a condition that evaluates to <code>true</code>.</p>
<p>Therefore, duplicating a condition automatically leads to dead code. Usually, this is due to a copy/paste error. At best, it’s simply dead code and
at worst, it’s a bug that is likely to induce further bugs as the code is maintained, and obviously it could lead to unexpected behavior.</p>
<h3>Noncompliant code example</h3>
<pre data-diff-id="1" data-diff-type="noncompliant">
if param == 1 {
  openWindow();
} else if param == 2 {
  closeWindow();
} else if param == 1 { // Noncompliant
  moveWindowToTheBackground();
}
</pre>
<h3>Compliant solution</h3>
<pre data-diff-id="1" data-diff-type="compliant">
if param == 1 {
  openWindow();
} else if param == 2 {
  closeWindow();
} else if param == 3 {
  moveWindowToTheBackground();
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#ifs_same_cond">https://rust-lang.github.io/rust-clippy/master/index.html#ifs_same_cond</a> </li>
</ul>

