<h2>Why is this an issue?</h2>
<p>Erasing operations (e.g., any operation involving multiplying by zero, dividing zero by a value, or bitwise AND with zero) always result in zero
regardless of the operands, which makes the expression unnecessary and can be replaced directly with zero. This simplifies the code and avoids
potential confusion about the intent of the expression.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
let x = 1;
let result = 0 * x; // Noncompliant: Result is always zero.
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
let x = 1;
let result = 0; // Compliant: Simplified expression.
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#erasing_op">https://rust-lang.github.io/rust-clippy/master/index.html#erasing_op</a>
  </li>
</ul>

