<h2>Why is this an issue?</h2>
<p>When using a floating-point <code>while</code> loop counter, an accumulation of rounding errors may result in a mismatch between the expected and
actual number of iterations.</p>
<p>Even if floating-point loop counters appears to behave correctly on one implementation, it may give a different number of iterations on another
implementation.</p>
<h2>How to fix it</h2>
<p>To fix the issue, use an integer loop counter instead. This avoids the accumulation of rounding errors and ensures that the loop iterates the
expected number of times.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
fn main() {
    let mut counter = 0.0;
    while counter &lt; 10.0 {
        println!("Counter: {}", counter);
        counter += 1.0;
    }
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
fn main() {
    let mut counter = 0;
    while counter &lt; 10 {
        println!("Counter: {}", counter);
        counter += 1;
    }
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#while_float">https://rust-lang.github.io/rust-clippy/master/index.html#while_float</a> </li>
</ul>

