<h2>Why is this an issue?</h2>
<p>Regular expressions have their own syntax that is understood by regular expression engines. Those engines will throw an exception at runtime if
they are given a regular expression that does not conform to that syntax.</p>
<p>To avoid syntax errors, special characters should be escaped with backslashes when they are intended to be matched literally and references to
capturing groups should use the correctly spelled name or number of the group.</p>
<h3>Noncompliant code example</h3>
<pre data-diff-id="1" data-diff-type="noncompliant">
Regex::new("(")
</pre>
<h3>Compliant solution</h3>
<pre data-diff-id="1" data-diff-type="compliant">
Regex::new("\(")
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Regex crate documentation - <a href="https://crates.io/crates/regex">regex</a> </li>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#invalid_regex">https://rust-lang.github.io/rust-clippy/master/index.html#invalid_regex</a> </li>
</ul>

