<h2>Why is this an issue?</h2>
<p>Having an immutable condition in a while loop can result in an infinite loop as the condition will never change, causing the loop to run
indefinitely.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
let i = 0;
while i &gt; 10 {
    println!("let me loop forever!"); // Noncompliant: 'i' does not change within the loop body
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
let mut i = 0;
while i &lt;= 10 {
    println!("looping: {}", i); // Compliant: 'i' gets incremented within the loop body
    i += 1;
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#while_immutable_condition">https://rust-lang.github.io/rust-clippy/master/index.html#while_immutable_condition</a> </li>
</ul>

