<h2>Why is this an issue?</h2>
<p>Performing operations like <code>offset</code> or <code>wrapping_add</code>/<code>wrapping_sub</code> on raw pointers to zero-sized types (ZSTs)
results in no-op operations. This is likely unintended and could make the code misleading, potentially hiding bugs.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
unsafe { (&amp;() as *const ()).offset(1); } // Noncompliant: No-op on zero-sized type pointer.
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
// Removing the no-op operation as it has no effect.
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#zst_offset">https://rust-lang.github.io/rust-clippy/master/index.html#zst_offset</a>
  </li>
</ul>

