<h2>Why is this an issue?</h2>
<p>Using <code>repr(isize/usize)</code> for C-like enumerations with values that donâ€™t fit into an <code>i32</code> can lead to truncated variant
values on 32-bit architectures, potentially causing unexpected behavior.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
#[repr(usize)]
enum NonPortable {
    X = 0x1_0000_0000, // Noncompliant
    Y = 0,
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
#[repr(u64)]
enum Portable {
    X = 0x1_0000_0000, // Compliant
    Y = 0,
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#enum_clike_unportable_variant">https://rust-lang.github.io/rust-clippy/master/index.html#enum_clike_unportable_variant</a> </li>
</ul>

