<h2>Why is this an issue?</h2>
<p>Hashing a unit value doesnâ€™t accomplish anything because the implementation of <code>Hash</code> for <code>()</code> is a no-op. This can lead to
confusion or misleading code, as it implies some hashing operation is actually taking place, when in reality it does nothing.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
# use std::hash::Hash;
# use std::collections::hash_map::DefaultHasher;
# enum Foo { Empty, WithValue(u8) }
# use Foo::*;
# let mut state = DefaultHasher::new();
# let my_enum = Foo::Empty;
match my_enum {
    Empty =&gt; ().hash(&amp;mut state), // Noncompliant
    WithValue(x) =&gt; x.hash(&amp;mut state),
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
# use std::hash::Hash;
# use std::collections::hash_map::DefaultHasher;
# enum Foo { Empty, WithValue(u8) }
# use Foo::*;
# let mut state = DefaultHasher::new();
# let my_enum = Foo::Empty;
match my_enum {
    Empty =&gt; 0_u8.hash(&amp;mut state), // Compliant
    WithValue(x) =&gt; x.hash(&amp;mut state),
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#unit_hash">https://rust-lang.github.io/rust-clippy/master/index.html#unit_hash</a>
  </li>
</ul>

