<h2>Why is this an issue?</h2>
<p>Redundant comparisons against constants results in always-true or always-false conditions. They add unnecessary complexity to the code and often
indicate a logical error where the developer might have intended a different comparison operator or a comparison with another value.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
let status_code = 200;
if status_code &lt;= 400 &amp;&amp; status_code &lt; 500 {} // Noncompliant: Redundant double comparison.
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
let status_code = 200;
if status_code &lt; 500 {} // Compliant: Corrected the comparison logic.
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#redundant_comparisons">https://rust-lang.github.io/rust-clippy/master/index.html#redundant_comparisons</a> </li>
</ul>

