<h2>Why is this an issue?</h2>
<p>When casting a raw pointer to a slice, the produced raw pointer does not correctly update its length metadata, leading to potential issues such as
creating a slice with less data than intended or more data, risking undefined behavior upon dereferencing.</p>
<h2>How to fix it</h2>
<p>Use <code>ptr::slice_from_raw_parts</code> to construct a slice from a data pointer and the correct length.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
let a = [1_u8, 2, 3, 4];
let p = &amp;a as *const [u8] as *const [u32];
unsafe {
    println!("{:?}", &amp;*p); // Noncompliant: Undefined behavior
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
let a = [1_i32, 2, 3, 4];
let old_ptr = &amp;a as *const [i32];
// The data pointer is cast to a pointer to the target `u8` not `[u8]`
// The length comes from the known length of 4 i32s times the 4 bytes per i32
let new_ptr = core::ptr::slice_from_raw_parts(old_ptr as *const u8, 16);
unsafe {
    println!("{:?}", &amp;*new_ptr);
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#cast_slice_different_sizes">https://rust-lang.github.io/rust-clippy/master/index.html#cast_slice_different_sizes</a> </li>
</ul>

