<h2>Why is this an issue?</h2>
<p>For any raw pointer passed as an argument, it is not possible to guarantee its validity. Dereferencing an invalid pointer can lead to undefined
behavior, causing potential segmentation faults or other critical issues. By marking such functions as <code>unsafe</code>, it notifies the caller
that they need to ensure the pointerâ€™s validity and surround the function call with an <code>unsafe</code> block to acknowledge the potential risks
involved.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
pub fn foo(x: *const u8) {
    println!("{}", unsafe { *x });
}

// This call "looks" safe but will segfault or worse!
// foo(invalid_ptr);
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
pub unsafe fn foo(x: *const u8) {
    println!("{}", unsafe { *x });
}

// This would cause a compiler error for calling without `unsafe`
// foo(invalid_ptr);

// Sound call if the caller knows the pointer is valid
unsafe { foo(valid_ptr); }
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#not_unsafe_ptr_arg_deref">https://rust-lang.github.io/rust-clippy/master/index.html#not_unsafe_ptr_arg_deref</a> </li>
</ul>

