<h2>Why is this an issue?</h2>
<p>Applying the <code>#[inline]</code> attribute to trait methods without provided implementations is pointless because only method implementations
can be inlined. The attribute is therefore ignored in such cases, leading to unnecessary code complexity.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
trait Animal {
    #[inline] // Noncompliant: Inline attribute on trait method without implementation.
    fn name(&amp;self) -&gt; &amp;'static str;
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
trait Animal {
    fn name(&amp;self) -&gt; &amp;'static str;
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#inline_fn_without_body">https://rust-lang.github.io/rust-clippy/master/index.html#inline_fn_without_body</a> </li>
</ul>

