<h2>Why is this an issue?</h2>
<p>Using <code>unwrap()</code> in conditions where it will always fail leads to unexpected panics and unclear error handling. Itâ€™s likely a mistake
and checking that there is indeed a wrapped value and not a none one was intended instead, which would require inverting the condition. However, if
panicking was indeed intended, then one should use <code>panic!</code> explicitly to make the intention clear and readable.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
let option = Some(0);
fn do_something_with(_x: usize) {}
if option.is_none() {
    do_something_with(option.unwrap()); // Noncompliant: This code will always panic.
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
let option = Some(0);
fn do_something_with(_x: usize) {}
if option.is_some() {
    do_something_with(option.unwrap()); // Compliant: Inverted condition to ensure unwrap is safe.
}

// or

if option.is_none() {
    panic!("Option was none when it was expected to be some."); // Compliant: Explicit panic with a message.
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#panicking_unwrap">https://rust-lang.github.io/rust-clippy/master/index.html#panicking_unwrap</a> </li>
</ul>

