<h2>Why is this an issue?</h2>
<p>Calling <code>.step_by(0)</code> on an iterator will cause the program to panic. This is likely an oversight and should be corrected to ensure
program stability. If the intent is to cause a panic, it is clearer to use <code>panic!()</code> directly.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
for x in (0..100).step_by(0) { // Noncompliant: This will cause a panic.
// ...
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
for x in (0..100).step_by(1) { // Compliant: Step by a valid positive integer.
// ...
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#iterator_step_by_zero">https://rust-lang.github.io/rust-clippy/master/index.html#iterator_step_by_zero</a> </li>
</ul>

