<h2>Why is this an issue?</h2>
<p>Using <code>.skip(0)</code> on an iterator does nothing, which is likely to be a mistake. It is more probable that <code>.skip(1)</code> was
intended to skip the first element, or if not, the <code>.skip(0)</code> call should be removed to avoid confusion and make the code clearer.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
let v = vec![1, 2, 3];
let x = v.iter().skip(0).collect::&lt;Vec&lt;_&gt;&gt;(); // Noncompliant: .skip(0) does nothing
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
let v = vec![1, 2, 3];
let x = v.iter().collect::&lt;Vec&lt;_&gt;&gt;(); // Compliant: No .skip(0) call makes the intent clear
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#iter_skip_zero">https://rust-lang.github.io/rust-clippy/master/index.html#iter_skip_zero</a> </li>
</ul>

