<h2>Why is this an issue?</h2>
<p>Defining an inherent <code>to_string(&amp;self)</code> method on a type that also implements the <code>Display</code> trait can lead to confusion,
as this inherent method will overshadow the <code>to_string</code> method automatically provided by the <code>Display</code> trait. The inherent
method is less versatile, preventing proper use of the <code>Display</code> traitâ€™s feature.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
use std::fmt;

pub struct A;

impl A {
    pub fn to_string(&amp;self) -&gt; String {
        "I am A".to_string() // Noncompliant: Inherent method shadows `Display::to_string`.
    }
}

impl fmt::Display for A {
    fn fmt(&amp;self, f: &amp;mut fmt::Formatter) -&gt; fmt::Result {
        write!(f, "I am A, too")
    }
}
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
use std::fmt;

pub struct A;

impl fmt::Display for A {
    fn fmt(&amp;self, f: &amp;mut fmt::Formatter) -&gt; fmt::Result {
        write!(f, "I am A")
    }
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#inherent_to_string_shadow_display">https://rust-lang.github.io/rust-clippy/master/index.html#inherent_to_string_shadow_display</a> </li>
</ul>

