<h2>Why is this an issue?</h2>
<p>Using conditional subtraction <code>if a &gt; b { b - a } else { 0 }</code> can lead to an unintended underflow, which can cause bugs or unexpected
behaviors. The <code>saturating_sub</code> method ensures that the subtraction does not underflow by returning zero if the result would have been
negative.</p>
<h3>Code examples</h3>
<h4>Noncompliant code example</h4>
<pre data-diff-id="1" data-diff-type="noncompliant">
let a = 12u32;
let b = 13u32;
let result = if a &gt; b { b - a } else { 0 }; // Noncompliant: Potential underflow condition.
</pre>
<h4>Compliant solution</h4>
<pre data-diff-id="1" data-diff-type="compliant">
let a = 12u32;
let b = 13u32;
let result = a.saturating_sub(b); // Compliant: Safe subtraction using saturating_sub.
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#inverted_saturating_sub">https://rust-lang.github.io/rust-clippy/master/index.html#inverted_saturating_sub</a> </li>
</ul>

