<h2>Why is this an issue?</h2>
<p>Using a <code>match</code> expression with a boolean condition is considered a bad practice because it adds unnecessary complexity to the code. The
<code>match</code> expression is designed for pattern matching and is more powerful and flexible than an <code>if-else</code> construct. However, when
used with a simple boolean condition, it introduces extra verbosity and reduces code readability. Instead, it is generally better to use an
<code>if-else</code> construct.</p>
<h3>Noncompliant code example</h3>
<pre data-diff-id="1" data-diff-type="noncompliant">
fn check_value(value: bool) -&gt; &amp;'static str {
    match value {
        true =&gt; "Value is true",
        false =&gt; "Value is false",
    }
}
</pre>
<h3>Compliant solution</h3>
<pre data-diff-id="1" data-diff-type="compliant">
fn check_value(value: bool) -&gt; &amp;'static str {
    if value {
        "Value is true"
    } else {
        "Value is false"
    }
}
</pre>
<h2>Resources</h2>
<h3>Documentation</h3>
<ul>
  <li> Clippy Lints - <a
  href="https://rust-lang.github.io/rust-clippy/master/index.html#match_bool">https://rust-lang.github.io/rust-clippy/master/index.html#match_bool</a>
  </li>
</ul>

